server/src/main/java/io/pravega/schemaregistry/server/rest/filter/NamespacePathRequestFilter.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.server.rest.filter;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.PathSegment;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+import java.util.List;
+
+/**
+ * A request filter that converts namespace expressed as a path param into a query param. 
+ * 
+ * This allows users to express either "/v1/namespace/ns/groups" "/v1/groups?namespace=ns" and both are equivalent. 
+ * 
+ * If both path param and query param are used to supply values for namespace, then the path param overrides the query param. 
+ * So if someone expresses the uri as "/v1/namespace/ns1/groups?namespace=ns2". This will result in namespace being 
+ * set to ns1 for the request.  
+ */
+@Provider
+@PreMatching
```

Comment:
This annotation implies we execute a filter before the resource matching. This implies the filter is invoked for every resource and not only for NamespacePath right ?

Reply:
Yes, its invoked for every request. 
Before looking at the request we dont know if it is for the namespace path or not. 
Its only in the request that we check if its the namespace path and extract the namespace from it.

Reply:
Would'nt this bypass `javax.ws.rs.core.SecurityContext` in the resource impl?

Reply:
Yes, you rightly pointed out that we use authentication and authorization in the resource implementation using rest auth handler. 
We actually dont use SecurityContext. 

But we could use it as a Permatching authentication filter and set its priority before the namespace filter so that all processing happens post authentication.
And then we can authorize in the Resource code. 

I have opened an issue for the same: #69